AK/StreamBuffer.h

```diff
@@ -124,8 +125,113 @@ class StreamBuffer {
     u8* m_data { nullptr };
 };
 
+class StreamSeekbackBuffer {
```

Comment:
Mainly because of performance (`CircularBuffer` is significantly slower than `StreamSeekbackBuffer`).

That said, I should see if I can factor out copying from seekback from here and reuse this faster implementation in `CircularBuffer`.

Reply:
> That said, I should see if I can factor out copying from seekback from here and reuse the faster implementation in CircularBuffer.

Would be nice (no matter the order), having the code in use in more places would make me significantly more confident that everything works as intended. (And having a matching before-after comparison would be cool too.)